### PR TITLE
[admission-policy-engine] Verify image signature

### LINKED ISSUE #10833: [admission-policy-engine] Verify image signature
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

We have to check image signature with public key or signature

### Proposed Solution

Add new policy to the [security policies](https://github.com/deckhouse/deckhouse/blob/ada1fde3f7b70a9c51201efbcd4c4c8218aa37a9/modules/015-admission-policy-engine/crds/security-policy.yaml#L96)

Public API should be something like:
```yaml
apiVersion: deckhouse.io/v1alpha1
kind: SecurityPolicy
metadata:
  name: mypolicy
spec:
  enforcementAction: Deny
  policies:
    verifyImages:
      - imageReferences:
          - "ghcr.io/repo/test-verify-image*" # use repo as string prefix or wildcard? I guess string prefix is better
        keys:
        - publicKeys: |-
                -----BEGIN PUBLIC KEY-----
                MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE8nXRh950IZbRj8Ra/N9sbqOPZrfM
                5/KAQN0/KjHcorm/J5yctVd7iEcnessRQjU917hmKO6JWVGHpDguIyakZA==
                -----END PUBLIC KEY-----                
          rekor:
             ignoreTlog: true
             url: https://rekor.sigstore.dev
```

### Additional Information

Should be something like [this](https://kyverno.io/docs/writing-policies/verify-images/sigstore/#verifying-image-signatures)

Possible solutions:
  - Rewrite outdated [provider](https://github.com/sigstore/cosign-gatekeeper-provider)
  - Integrate [ratify](https://ratify.dev/)

### PR DESCRIPTION
**Pull request by miklezzzz**
## Description

Provides a way of enforcing container image signature policies (cosign compliant). 

## Why do we need it, and what problem does it solve?
The pr introduces a new setting to security policies - verifyImageSignatures, that enables a user to implement container image signature check policies (way of checking that an image, running in the cluster, is signed with a correct cosign signature).
The functionality is available in the EE/FE/CSE Deckhouse editions.
Closes #10833

## What is the expected result?
policy.yaml
```yaml
apiVersion: deckhouse.io/v1alpha1
kind: SecurityPolicy
metadata:
  name: example
spec:
  match:
    namespaceSelector:
      labelSelector:
        matchLabels:
          kubernetes.io/metadata.name: default
  policies:
    verifyImageSignatures:
      - reference: docker.io/myrepo/*
        publicKeys:
        - |-
          -----BEGIN PUBLIC KEY-----
          MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE8nXRh950IZbRj8Ra/N9sbqOPZrfM
          5/KAQN0/KjHcorm/J5yctVd7iEcnessRQjU917hmKO6JWVGHpDguIyakZA==
          -----END PUBLIC KEY-----
      - reference: company.registry.com/*
        dockerCfg: zxc==
        publicKeys:
        - |-
          -----BEGIN PUBLIC KEY-----
          MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE8nXRh950IZbRj8Ra/N9sbqOPZrfM
          5/KAQN0/KjHcorm/J5yctVd7iEcnessRQjU917hmKO6JWVGHpDguIyakZA==
          -----END PUBLIC KEY-----
```

create unsigned pod
```shell
Error from server (Forbidden): error when creating "unsigned.yaml": admission webhook "admission-policy-engine.deckhouse.io" denied the request: [example] Image signature verification failed: nginx:1.14.2
[example] Image signature verification failed: nginx:1.17.2
```

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: admission-policy-engine
type: feature 
summary: Add SecurityPolicy with ability to check images, signed with cosign. EE only.
impact_level: default 
```

### CODE REVIEW

File: modules/015-admission-policy-engine/templates/audit-deployment.yaml [hunk 1]
```diff
@@ -81,6 +81,7 @@ spec:
         - --enable-external-data=true
         - --metrics-backend=prometheus
         - --disable-cert-rotation=true
+        - --external-data-provider-response-cache-ttl=30s
```
Comment (yalosev) [inline modules/015-admission-policy-engine/templates/audit-deployment.yaml:84]: TODO: remove

File: modules/015-admission-policy-engine/templates/gatekeeper-deployment.yaml [hunk 2]
```diff
@@ -76,6 +76,7 @@ spec:
         - --disable-cert-rotation=true
         - --metrics-backend=prometheus
         - --disable-opa-builtin={http.send}
+        - --external-data-provider-response-cache-ttl=30s
```
Comment (yalosev) [inline modules/015-admission-policy-engine/templates/gatekeeper-deployment.yaml:79]: TODO: remove

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/stage'...

**Review by z9r5 — APPROVED** [review]

**Review by yalosev — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by yalosev** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.27 12116001756 12116001756-1-con-1-27 /sys/deckhouse-oss/install:pr10846 redos@158.160.55.103

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Failure: cherry-pick commit 1835b79c846fa45f2f2487a22ff3948f7abaa4ba to the branch [release-1.66](https://github.com/deckhouse/deckhouse/tree/release-1.66) failed. See [Job](https://github.com/deckhouse/deckhouse/actions/runs/12117517547) for details.

**Pull request merged**